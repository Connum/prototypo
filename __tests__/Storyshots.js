import initStoryshots from '@storybook/addon-storyshots';

// This extra is needed to avoid errors due to React 16
global.window = global;
window.addEventListener = () => {};
window.requestAnimationFrame = () => {
	throw new Error('requestAnimationFrame is not supported in Node');
};

initStoryshots();
