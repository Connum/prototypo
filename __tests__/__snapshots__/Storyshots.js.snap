exports[`Storyshots SliderController 50% filled 1`] = `
<div
  style={
    Object {
      "backgroundColor": "#3b3b3b",
      "bottom": 0,
      "left": 0,
      "position": "absolute",
      "right": 0,
      "top": 0,
      "transition": "background-color 0.25s ease-in-out",
    }
  }>
  <div
    className="normal"
    style={
      Object {
        "margin": "auto",
        "padding": "20px",
        "width": "300px",
      }
    }>
    <div
      className="slider-controller"
      onMouseDown={[Function]}>
      <div
        className="slider-controller-bg"
        style={
          Object {
            "transform": "translateX(-46%)",
          }
        }>
        <div
          className="slider-controller-handle" />
      </div>
    </div>
  </div>
  <ul
    style={
      Object {
        "bottom": "10px",
        "margin": 0,
        "overflow": "auto",
        "position": "absolute",
        "right": "10px",
      }
    }>
    <li
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "#3b3b3b",
          "borderRadius": "4px",
          "boxShadow": "inset 0 0 0 2px rgba(0, 0, 0, 0.5)",
          "cursor": "pointer",
          "display": "block",
          "float": "left",
          "height": "20px",
          "margin": "0 0 0 10px",
          "transition": "opacity 0.25s ease-in-out",
          "width": "20px",
        }
      } />
  </ul>
</div>
`;

exports[`Storyshots SliderController empty 1`] = `
<div
  style={
    Object {
      "backgroundColor": "#3b3b3b",
      "bottom": 0,
      "left": 0,
      "position": "absolute",
      "right": 0,
      "top": 0,
      "transition": "background-color 0.25s ease-in-out",
    }
  }>
  <div
    className="normal"
    style={
      Object {
        "margin": "auto",
        "padding": "20px",
        "width": "300px",
      }
    }>
    <div
      className="slider-controller"
      onMouseDown={[Function]}>
      <div
        className="slider-controller-bg"
        style={
          Object {
            "transform": "translateX(-92%)",
          }
        }>
        <div
          className="slider-controller-handle" />
      </div>
    </div>
  </div>
  <ul
    style={
      Object {
        "bottom": "10px",
        "margin": 0,
        "overflow": "auto",
        "position": "absolute",
        "right": "10px",
      }
    }>
    <li
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "#3b3b3b",
          "borderRadius": "4px",
          "boxShadow": "inset 0 0 0 2px rgba(0, 0, 0, 0.5)",
          "cursor": "pointer",
          "display": "block",
          "float": "left",
          "height": "20px",
          "margin": "0 0 0 10px",
          "transition": "opacity 0.25s ease-in-out",
          "width": "20px",
        }
      } />
  </ul>
</div>
`;

exports[`Storyshots SliderController full 1`] = `
<div
  style={
    Object {
      "backgroundColor": "#3b3b3b",
      "bottom": 0,
      "left": 0,
      "position": "absolute",
      "right": 0,
      "top": 0,
      "transition": "background-color 0.25s ease-in-out",
    }
  }>
  <div
    className="normal"
    style={
      Object {
        "margin": "auto",
        "padding": "20px",
        "width": "300px",
      }
    }>
    <div
      className="slider-controller"
      onMouseDown={[Function]}>
      <div
        className="slider-controller-bg"
        style={
          Object {
            "transform": "translateX(-0%)",
          }
        }>
        <div
          className="slider-controller-handle" />
      </div>
    </div>
  </div>
  <ul
    style={
      Object {
        "bottom": "10px",
        "margin": 0,
        "overflow": "auto",
        "position": "absolute",
        "right": "10px",
      }
    }>
    <li
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "#3b3b3b",
          "borderRadius": "4px",
          "boxShadow": "inset 0 0 0 2px rgba(0, 0, 0, 0.5)",
          "cursor": "pointer",
          "display": "block",
          "float": "left",
          "height": "20px",
          "margin": "0 0 0 10px",
          "transition": "opacity 0.25s ease-in-out",
          "width": "20px",
        }
      } />
  </ul>
</div>
`;

exports[`Storyshots SliderController restricted 1`] = `
<div
  style={
    Object {
      "backgroundColor": "#3b3b3b",
      "bottom": 0,
      "left": 0,
      "position": "absolute",
      "right": 0,
      "top": 0,
      "transition": "background-color 0.25s ease-in-out",
    }
  }>
  <div
    className="normal"
    style={
      Object {
        "margin": "auto",
        "padding": "20px",
        "width": "300px",
      }
    }>
    <div
      className="slider-controller"
      onMouseDown={[Function]}>
      <div
        className="slider-controller-bg"
        style={
          Object {
            "transform": "translateX(-27.599999999999998%)",
          }
        }>
        <div
          className="slider-controller-handle" />
      </div>
      <span>
        <div
          className="slider-range-limiter-bg min"
          onClick={[Function]}
          style={
            Object {
              "marginLeft": "-10px",
              "transform": "translateX(-36.800000000000004%)",
            }
          } />
        <div
          className="slider-range-limiter-bg max"
          onClick={[Function]}
          style={
            Object {
              "marginLeft": "-10px",
              "transform": "translateX(81.6%)",
            }
          } />
      </span>
    </div>
  </div>
  <ul
    style={
      Object {
        "bottom": "10px",
        "margin": 0,
        "overflow": "auto",
        "position": "absolute",
        "right": "10px",
      }
    }>
    <li
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "#3b3b3b",
          "borderRadius": "4px",
          "boxShadow": "inset 0 0 0 2px rgba(0, 0, 0, 0.5)",
          "cursor": "pointer",
          "display": "block",
          "float": "left",
          "height": "20px",
          "margin": "0 0 0 10px",
          "transition": "opacity 0.25s ease-in-out",
          "width": "20px",
        }
      } />
  </ul>
</div>
`;

exports[`Storyshots SliderController unadvised state 1`] = `
<div
  style={
    Object {
      "backgroundColor": "#3b3b3b",
      "bottom": 0,
      "left": 0,
      "position": "absolute",
      "right": 0,
      "top": 0,
      "transition": "background-color 0.25s ease-in-out",
    }
  }>
  <div
    className="normal"
    style={
      Object {
        "margin": "auto",
        "padding": "20px",
        "width": "300px",
      }
    }>
    <div
      className="slider-controller"
      onMouseDown={[Function]}>
      <div
        className="slider-controller-bg is-not-advised"
        style={
          Object {
            "transform": "translateX(-4.6000000000000005%)",
          }
        }>
        <div
          className="slider-controller-handle" />
      </div>
    </div>
  </div>
  <ul
    style={
      Object {
        "bottom": "10px",
        "margin": 0,
        "overflow": "auto",
        "position": "absolute",
        "right": "10px",
      }
    }>
    <li
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "#3b3b3b",
          "borderRadius": "4px",
          "boxShadow": "inset 0 0 0 2px rgba(0, 0, 0, 0.5)",
          "cursor": "pointer",
          "display": "block",
          "float": "left",
          "height": "20px",
          "margin": "0 0 0 10px",
          "transition": "opacity 0.25s ease-in-out",
          "width": "20px",
        }
      } />
  </ul>
</div>
`;

exports[`Storyshots account.AccountManageSubUsers empty state 1`] = `
<div
  className="display-with-label undefined">
  <label
    className="display-with-label-label" />
  <div
    className="display-with-label-display">
    <p>
      You don\'t manage any user for now.
    </p>
  </div>
</div>
`;

exports[`Storyshots account.AccountManageSubUsers empty state editable 1`] = `
<div
  className="display-with-label undefined">
  <label
    className="display-with-label-label" />
  <div
    className="display-with-label-display">
    <p>
      You don\'t manage any user for now.
    </p>
    <ul>
      <li
        style={
          Object {
            "listStyleType": "none",
          }
        }>
        <form
          onSubmit={[Function]}>
          <input
            name="email"
            placeholder="email@example.com"
            type="text" />
          <button>
            Add user
          </button>
        </form>
      </li>
    </ul>
  </div>
</div>
`;

exports[`Storyshots account.AccountManageSubUsers full 1`] = `
<div
  className="display-with-label undefined">
  <label
    className="display-with-label-label" />
  <div
    className="display-with-label-display">
    <ul>
      <li>
        user@example.com
         
      </li>
      <li>
        user2@example.com
         
      </li>
    </ul>
    <p>
      0
       slots left.
    </p>
  </div>
</div>
`;

exports[`Storyshots account.AccountManageSubUsers removable 1`] = `
<div
  className="display-with-label undefined">
  <label
    className="display-with-label-label" />
  <div
    className="display-with-label-display">
    <ul>
      <li>
        user@example.com
         
        <button
          onClick={[Function]}>
          ×
        </button>
      </li>
      <li>
        user2@example.com
         
        <button
          onClick={[Function]}>
          ×
        </button>
      </li>
    </ul>
    <p>
      0
       slots left.
    </p>
  </div>
</div>
`;

exports[`Storyshots account.AccountManageSubUsers with 2 people 1`] = `
<div
  className="display-with-label undefined">
  <label
    className="display-with-label-label" />
  <div
    className="display-with-label-display">
    <ul>
      <li>
        user@example.com
         
      </li>
      <li>
        user2@example.com
         
      </li>
      <li
        style={
          Object {
            "listStyleType": "none",
          }
        }>
        <form
          onSubmit={[Function]}>
          <input
            name="email"
            placeholder="email@example.com"
            type="text" />
          <button>
            Add user
          </button>
        </form>
      </li>
    </ul>
  </div>
</div>
`;

exports[`Storyshots account.AccountManageSubUsers with limited quantity 1`] = `
<div
  className="display-with-label undefined">
  <label
    className="display-with-label-label" />
  <div
    className="display-with-label-display">
    <ul>
      <li>
        user@example.com
         
      </li>
      <li>
        user2@example.com
         
      </li>
      <li
        style={
          Object {
            "listStyleType": "none",
          }
        }>
        <form
          onSubmit={[Function]}>
          <input
            name="email"
            placeholder="email@example.com"
            type="text" />
          <button>
            Add user
          </button>
        </form>
      </li>
    </ul>
    <p>
      3
       slots left.
    </p>
  </div>
</div>
`;

exports[`Storyshots shared.AccountValidationButton default 1`] = `
<button
  className="account-button account-validation-button"
  disabled={undefined}
  onClick={undefined}
  type="submit">
  <div>
    Hit me!
  </div>
</button>
`;

exports[`Storyshots shared.AccountValidationButton disabled 1`] = `
<button
  className="account-button account-validation-button"
  disabled={true}
  onClick={undefined}
  type="submit">
  <div>
    You can\'t hit me!
  </div>
</button>
`;

exports[`Storyshots shared.AccountValidationButton loading 1`] = `
<button
  className="account-button account-validation-button"
  disabled={true}
  onClick={undefined}
  type="submit">
  <div>
    <div
      className="wait-for-load">
      <div
        className="sk-spinner sk-spinner-wave sk-secondary-color">
        <div
          className="sk-rect1" />
        <div
          className="sk-rect2" />
        <div
          className="sk-rect3" />
        <div
          className="sk-rect4" />
        <div
          className="sk-rect5" />
      </div>
    </div>
  </div>
</button>
`;
